syntax = "proto3";

package collector;

service Collector {
  rpc SendEvent(Metric) returns (Ack) {}

  rpc SendMetric(Metric) returns (Ack) {}
  rpc SendMetricStream(stream Metric) returns (Ack) {}
}


message Metric {
   string domain = 1;
   string opco = 2;
   string customer = 3;
   string resource = 4;
   uint32 messageVersion = 5;

   uint64 eventEpoc = 6;


  message Value {
     string name = 1;
     float value = 2;
  }

  message Tag {
     string name = 1;
     string value = 2;
  }

  repeated Value values = 7;
  repeated Tag tags = 8;


}


message Ack {
  int32 code = 1;
}